go-controller/pkg/util/pod_annotation.go

```diff
@@ -532,22 +572,24 @@ func AddRoutesGatewayIP(
 			}
 		}
 
-		// Ensure default service network traffic always goes to OVN
-		for _, serviceSubnet := range config.Kubernetes.ServiceCIDRs {
-			if isIPv6 == utilnet.IsIPv6CIDR(serviceSubnet) {
-				podAnnotation.Routes = append(podAnnotation.Routes, PodRoute{
-					Dest:    serviceSubnet,
-					NextHop: gatewayIPnet.IP,
-				})
+		if podAnnotation.Primary {
+			// Ensure default service network traffic always goes to OVN
+			for _, serviceSubnet := range config.Kubernetes.ServiceCIDRs {
+				if isIPv6 == utilnet.IsIPv6CIDR(serviceSubnet) {
+					podAnnotation.Routes = append(podAnnotation.Routes, PodRoute{
+						Dest:    serviceSubnet,
+						NextHop: gatewayIPnet.IP,
+					})
+				}
```

Comment:
Following the three rule, since this pattern appears three times at the file, it would be a good idea to wrap it at a function.

Reply:
agreed :)

Reply:
done!